demisto_sdk/commands/setup_env/setup_environment.py

```diff
@@ -0,0 +1,406 @@
+) -> Path:
+    docker_client = docker_helper.init_global_docker_client()
+    requirements = (
+        docker_client.containers.run(
+            test_docker_image, command="pip list --format=freeze", remove=True
+        )
+        .decode()
+        .split("\n")
+    )
+    venv_path = integration_script.path.parent / "venv"
+    interpreter_path = venv_path / "bin" / "python"
+    if venv_path.exists() and not overwrite_virtualenv:
+        return interpreter_path
+    logger.info(f"Creating virtualenv for {integration_script.name}")
+    shutil.rmtree(venv_path, ignore_errors=True)
+    venv.create(venv_path, with_pip=True)
+    for req in requirements:
+        try:
+            if not req:
+                continue
+            subprocess.run(
+                [
+                    f"{venv_path / 'bin' / 'pip'}",
+                    "-q",
+                    "--disable-pip-version-check",
+                    "install",
+                    req,
+                ],
+                check=True,
+            )
+            logger.info(f"Installed {req}")
+        except subprocess.CalledProcessError:
+            logger.warning(f"Could not install {req}, skipping...")
+    return interpreter_path
+
+
+def configure_params(
+    integration_script: IntegrationScript,
+    secret_id: Optional[str],
+    instance_name: Optional[str],
+):
+    """Configuring the integration parameters locally and in XSOAR/XSIAM
+
+    Args:
+        integration_script (IntegrationScript): The integration script object configure params to
+        secret_id (Optional[str]): The secret id of the parameters. defaults to the inegration name.
+        instance_name (Optional[str]): The instance name to configure on XSOAR/XSIAM. If None, will not configure.
+    """
+
+    if not secret_id:
+        secret_id = integration_script.name.replace(" ", "_")
+        secret_id = re.sub(r"[()]", "", secret_id)
+    if (project_id := os.getenv("DEMISTO_SDK_GCP_PROJECT_ID")) and isinstance(
+        integration_script, Integration
+    ):
+        params = get_integration_params(project_id, secret_id)
+        if params and instance_name:
+            if (
+                instance_created := create_integration_instance(
+                    integration_script.name,
+                    instance_name,
+                    params,
+                    params.get("byoi", True),
+                )
+            ) and instance_created[0]:
+                logger.info(
+                    f"Created integration instance {instance_created[0]['name']}"
+                )
+            else:
+                logger.warning(f"Failed to create integration instance {instance_name}")
+        (CONTENT_PATH / ".vscode").mkdir(exist_ok=True)
+        with open(CONTENT_PATH / ".vscode" / "params.json", "w") as f:
+            json.dump(params, f, indent=4)
+    else:
+        logger.info(
+            "Skipping searching in Google Secret Manager as DEMISTO_SDK_GCP_PROJECT_ID is not set"
```

Comment:
Can the user do anything with this command when the `PROJECT_ID` isn't configured?

Reply:
If it's not configured, it will configure the local IDE, but without the secrets and the instance with XSOAR (pretty much what the extension does currently)